bbc/simorgh
# WSTEAM1-832+833 - SMP settings builder

Resolves JIRA  https://jira.dev.bbc.co.uk/browse/WSTEAM1-832 and https://jira.dev.bbc.co.uk/browse/WSTEAM1-833

Overall changes
======
- Creates SMP settings builder functions based on the page type requested
- Sets a 'base' configuration which all page types inherit and can then override in their setting builder function
- Note this work is mainly for the Live pages, however there is 'article' and 'mediaArticle' logic here. This is somewhat to test the theory of this config builder approach, as we'll need to do something like this in the near future as we move other page types over to using the new media player

Code changes
======
- Created settings builder functions for live and article page types. These are mostly copies of these functions: https://github.com/bbc/simorgh/blob/ce143285e1c0f2da86f09c7cb26d7427a5058bae/src/app/legacy/containers/MediaPlayer/helpers/propsInference/livePage/index.js with settings stripped back to just what we need for now. These will be extended in separate tickets/PRs as they come about
- Sets a base configuration that is then augmented with settings for the specific page type

Testing
======
1. Visit http://localhost.bbc.com:7081/pidgin/live/c7p765ynk9qt?renderer_env=test
2. Confirm the media player loads when the temporary 'placeholder' button is clicked
3. Ensure the media plays back as expected in the Toucan UI

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
